detect_secrets/filters/allowlist.py

```diff
@@ -2,100 +2,83 @@
+        ('--', ''),                   # e.g. sql
+        (r'<!--[# \t]*?', ' *?-->'),  # e.g. xml
+        # many other inline comment syntaxes are not included,
+        # because we want to be performant for
+        # any(regex.search(line) for regex in ALLOWLIST_REGEXES)
+        # calls. of course, this won't be a concern if detect-secrets
+        # switches over to implementing file plugins for each supported
+        # filetype.
+    )
+
+
+def _get_allowlist_regexes_for_file(filename: str) -> Iterable[List[Pattern]]:
+    comment_tuples = _get_comment_tuples()
 
+    _, ext = os.path.splitext(filename)
+    if ext[1:] in _get_file_to_index_dict():
+        comment_tuples = (comment_tuples[_get_file_to_index_dict()[ext[1:]]],)
 
-@lru_cache(maxsize=1)
-def _get_file_based_allowlist_nextline_regexes() -> Dict[str, List[Pattern]]:
-    # Add to this mapping (and ALLOWLIST_REGEXES if applicable) lazily,
-    # as more language specific file parsers are implemented.
-    # Discussion: https://github.com/Yelp/detect-secrets/pull/105
-    return {
-        'yaml': [_get_allowlist_nextline_regexes()[0]],
-    }
+    yield [
+        _get_allowlist_regexes(comment_tuple=t, nextline=False)
+        for t in comment_tuples
+    ]
+    yield [
+        _get_allowlist_regexes(comment_tuple=t, nextline=True)
+        for t in comment_tuples
+    ]
 
 
-@lru_cache(maxsize=1)
-def _get_allowlist_nextline_regexes() -> List[Pattern]:
-    return [
-        re.compile(r)
-        for r in [
-            r'^[ \t]*{} *pragma: ?allowlist[ -]nextline[ -]secret.*?{}[ \t]*$'.format(start, end)
-            for start, end in (
-                ('#', ''),                    # e.g. python or yaml
-                ('//', ''),                   # e.g. golang
-                (r'/\*', r' *\*/'),           # e.g. c
-                ('\'', ''),                   # e.g. visual basic .net
-                ('--', ''),                   # e.g. sql
-                (r'<!--[# \t]*?', ' *?-->'),  # e.g. xml
-                # many other inline comment syntaxes are not included,
-                # because we want to be performant for
-                # any(regex.search(line) for regex in ALLOWLIST_REGEXES)
-                # calls. of course, this won't be a concern if detect-secrets
-                # switches over to implementing file plugins for each supported
-                # filetype.
-            )
-        ]
-    ]
+# Note: Cache size should be 2x the number of comment types
+@lru_cache(maxsize=12)
+def _get_allowlist_regexes(comment_tuple: Tuple[str, str], nextline: bool) -> Pattern:
+    start = comment_tuple[0]
+    end = comment_tuple[1]
+    return re.compile(
+        # Note: Always use allowlist, whitelist will be deprecated in the future
+        r'{}{} *pragma: ?{}{}[ -]secret.*?{}[ \t]*$'.format(
+            r'^[ \t]*' if nextline else r'[ \t]+',
+            start,
+            r'allowlist' if nextline else r'(allow|white)list',
```

Comment:
I'm also fine if these are the same. Supporting `whitelist` is just for backwards compatibility, and I'm fine with keeping that backwards compatibility around for the nextline regex too.

Reply:
I don't mind one way or the other, but my thinking was that, b/c `nextline` is new, it doesn't have any backward compatibility to contend with. Thus, requiring `allowlist` for `nextline` prepares uses of it for the future removal of `whitelist`, whereas allowing both opens the possibility of further entrenching `whitelist`.